rllib/models/experimental/catalog.py

```diff
@@ -0,0 +1,99 @@
+from ray.rllib.models.experimental.configs import (
+    MLPModelConfig,
+    MLPEncoderConfig,
+    LSTMEncoderConfig,
+)
+from ray.rllib.models.experimental.base import ModelConfig
+from ray.rllib.models.catalog import MODEL_DEFAULTS
+from gymnasium.spaces import Box
+
+
+class Catalog:
+    """Defines how what models an RLModules builds.
+
+    RLlib's native RLModules get their Models from a ModelBuilder object.
+    By default, that ModelBuilder builds the configs it holds.
+    You can modify the ModelBuilder so that it builds different Models by subclassing
+    and don't have to write configs.
+    """
+
+    def __init__(self, observation_space, action_space, model_config):
+        self.observation_space = observation_space
+        self.action_space = action_space
+        # TODO (Artur): Possibly get rid of this config merge
+        self.model_config = {**MODEL_DEFAULTS, **model_config}
+        self.encoder_config = self.get_encoder_config(
+            observation_space, self.model_config
+        )
+        self.latent_dim = self.encoder_config.output_dim
+
+    def build_encoder(self, framework="torch"):
+        return self.encoder_config.build(framework=framework)
+
+    @staticmethod
+    def get_encoder_config(observation_space, model_config) -> ModelConfig:
```

Comment:
type hints and documentation plz on args, etc.

Reply:
Yep, adding that everywhere 🙂